cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -2644,7 +2649,8 @@ private ApplicationMeta decodeRow(StructuredRow row) {
     } else {
       changeDetail = new ChangeDetail(changeSummary, null, author, creationTimeMillis, latest);
     }
-    return new ApplicationMeta(id, spec, changeDetail, sourceControl, latest);
+
+    return new ApplicationMeta(id, spec, changeDetail, sourceControl, Boolean.TRUE.equals(latest));
```

Comment:
This is incorrect. If some caller directly uses this decoderow() they will get invalid data. Just resetting the value in getlatest will not solve the issue

Reply:
Ok I understand now what is reason for setting this. The latest null can be false if the row is not actually the last created. Hmm in that case I do not have any good suggestion. 
Maybe making latest Nullable makes sense here.

Reply:
latest will be null for rows that were inserted before pipeline versioning was introduced, which is the case when an instance has been upgraded from an older version to the newest version.

It's not possible to determine if this is the latest one just by looking at the row in that scenario, which is why the getLatest() method exists. I think it would be cleaner to have another `decodeRow(StructuredRow row, boolean isLatest)` method that takes a boolean for whether it is the latest or not. Change the getLatest() method to call it with true, and change this method to call it with the value from the row, which we will expect to never be null.

Reply:
Agree that we should pass the `isLatest` to `decodeRow`. Please chime in here as well @sanjanasandeep

Reply:
Agreed. Adding another `decodeRow` method seems to be a cleaner solution.

Reply:
Though after the refactors, this change here has been removed (no longer needed).